packages/mdc-shape/_functions.scss

```diff
@@ -147,34 +132,50 @@
     @error "Expected masked-corners of length 4 but got '#{length($masked-corners)}'.";
   }
 
-  @if length($radius) == 3 {
-    $radius: nth($radius, 1) nth($radius, 2) nth($radius, 3) nth($radius, 2);
-  } @else if length($radius) == 2 {
-    $radius: nth($radius, 1) nth($radius, 2) nth($radius, 1) nth($radius, 2);
-  } @else if length($radius) == 1 {
-    $radius: $radius $radius $radius $radius;
-  }
+  $radius: mdc-shape-unpack-radius_($radius);
 
   @return if(nth($masked-corners, 1) == 1, nth($radius, 1), 0)
     if(nth($masked-corners, 2) == 1, nth($radius, 2), 0)
     if(nth($masked-corners, 3) == 1, nth($radius, 3), 0)
     if(nth($masked-corners, 4) == 1, nth($radius, 4), 0);
 }
 
-@function mdc-shape-prop-corner-value_($radius) {
-  @if map-has-key($mdc-shape-category-values, $radius) {
-    @return map-get($mdc-shape-category-values, $radius);
-  } @else if mdc-shape-is-valid-radius-value_($radius) {
+//
+// Unpacks shorthand values for border-radius (i.e. lists of 1-3 values).
+// If a list of 4 values is given, it is returned as-is.
+// TODO: This is private for purposes of getting it into a patch; make it public for a future minor/major release.
+//
+@function mdc-shape-unpack-radius_($radius) {
+  @if length($radius) == 4 {
     @return $radius;
-  } @else {
-    @error "Invalid radius: '#{$radius}' radius is not supported";
+  } @else if length($radius) == 3 {
+    @return nth($radius, 1) nth($radius, 2) nth($radius, 3) nth($radius, 2);
+  } @else if length($radius) == 2 {
+    @return nth($radius, 1) nth($radius, 2) nth($radius, 1) nth($radius, 2);
+  } @else if length($radius) == 1 {
+    @return $radius $radius $radius $radius;
   }
 
-  @return map-get($mdc-shape-category-values, $radius);
+  @error "Invalid radius: '#{$radius}' is more than 4 values";
 }
 
-@function mdc-shape-is-valid-radius-value_($radius) {
+@function mdc-shape-resolve-percentage-for-corner_($component-height, $radius) {
+  @if type-of($radius) == "number" and unit($radius) == "%" {
+    // Converts the percentage to number without unit. Example: 50% => 50.
+    $percentage: $radius / ($radius * 0 + 1);
+
+    @return $component-height * ($percentage / 100);
+  } @else {
+    @return $radius;
+  }
+}
+
+@function mdc-shape-validate-radius-value_($radius) {
```

Comment:
This function has been rewritten to emit an error, akin to the `verify` functions in `feature-targeting`, so that there's less code to write in places that call this.